lib/ansible/modules/packaging/os/redhat_repositories.py

```diff
@@ -0,0 +1,399 @@
+        l={}
+        for repo in self.filter(stuff):
+            id=repo.id
+            id=id.replace("-", "_")
+            l[id] = dict(id=repo.id, name=repo.name, enabled=repo.is_enabled, url=repo.url)
+        return l
+
+    def enable_only(self, id):
+        changed=False
+        to_enable=[]
+        to_disable=[]
+
+        # no repo id => disable all
+        if len(id) == 0:
+            self.disable_all()
+            changed=True
+            return [ changed, dict(enabled=[], disabled='*') ]
+
+        # where are we now?
+        current_repos=self.list_stuff('all')
+
+        notice("redhat_repository: enable_only: changed=" + str(current_repos))
+        for r in current_repos.keys():
+            notice("redhat_repository: enable_only: **loop** id=" + str(id) + " r=" + str(r))
+
+            if current_repos[r]['enabled'] and current_repos[r]['id'] not in id:
+                to_disable.append(current_repos[r]['id'])
+                changed=True
+
+            if not current_repos[r]['enabled'] and current_repos[r]['id'] in id:
+                to_enable.append(current_repos[r]['id'])
+                changed=True
+
+        notice("redhat_repository: enable_only: changed=" + str(changed))
+        notice("redhat_repository: enable_only: to_enable=" + str(to_enable))
+        notice("redhat_repository: enable_only: to_disable=" + str(to_disable))
+        if changed:
+            args="subscription-manager repos "
+            for n in to_disable:
+                args += " --disable=" + n
+
+            for n in to_enable:
+                args+=" --enable=" + n
+
+            notice("redhat_repository: enable_only repos args: " + str(args))
+            rc, stdout, stderr = self.module.run_command(args, check_rc=True)
+
+        return [ changed, dict(enabled=to_enable,disabled=to_disable) ]
+
+    def enablerepo(self, id):
+        if len(id) > 0:
+            args="subscription-manager repos "
+            for n in id:
+                args+=" --enable=" + n
+            notice("redhat_repository: enable repos args: " + str(args))
+            rc, stdout, stderr = self.module.run_command(args, check_rc=True)
+            return True
+        else:
+            return False
+
+    def disablerepo(self, id):
+        if len(id) > 0:
+            args="subscription-manager repos "
+            for n in id:
+                args += " --disable=" + n
+            rc, stdout, stderr = self.module.run_command(args, check_rc=True)
+            return True
+        else:
+            return False
+
+    def disable_all(self):
+        notice("redhat_repository: disable all repos")
+        args = "subscription-manager repos --disable='*'"
+        rc, stdout, stderr = self.module.run_command(args, check_rc=True)
+
+def main():
+    #
+    # Initialise Ansible module
+    #
+    module = AnsibleModule(
+                argument_spec = dict(
+                    id      = dict(default=None, type="list"),
```

Comment:
The list of option could be ordered alphabetically. The `default=None` is not required.